plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoSession.java

```diff
@@ -210,6 +210,26 @@ public void addColumn(SchemaTableName schemaTableName, ColumnMetadata columnMeta
         tableCache.invalidate(schemaTableName);
     }
 
+    public void dropColumn(SchemaTableName schemaTableName, String columnName)
+    {
+        String schemaName = toRemoteSchemaName(schemaTableName.getSchemaName());
+        String tableName = toRemoteTableName(schemaName, schemaTableName.getTableName());
+
+        Document metadata = getTableMetadata(schemaName, tableName);
+
+        List<Document> columns = getColumnMetadata(metadata).stream()
+                .filter(doc -> !doc.getString(FIELDS_NAME_KEY).equals(columnName))
+                .collect(toImmutableList());
+
+        metadata.append(FIELDS_KEY, columns);
```

Comment:
does "append" replace the actual columns?

Reply:
Yes, the internal implementation is `LinkedHashMap.put()` method.